src/components/YieldPools/ElasticCompensation.tsx

```diff
@@ -0,0 +1,286 @@
+import styled from 'styled-components'
+
+import { ButtonLight, ButtonPrimary } from 'components/Button'
+import CurrencyLogo from 'components/CurrencyLogo'
+import DoubleCurrencyLogo from 'components/DoubleLogo'
+import Modal from 'components/Modal'
+import TransactionConfirmationModal, { TransactionErrorContent } from 'components/TransactionConfirmationModal'
+import { ELASTIC_BASE_FEE_UNIT, ZERO_ADDRESS } from 'constants/index'
+import { NativeCurrencies } from 'constants/tokens'
+import { useActiveWeb3React, useWeb3React } from 'hooks'
+import { Snapshot } from 'hooks/useElasticCompensationData'
+import useTheme from 'hooks/useTheme'
+import { useElasticFarms } from 'state/farms/elastic/hooks'
+import { useTransactionAdder } from 'state/transactions/hooks'
+import { TRANSACTION_TYPE } from 'state/transactions/type'
+import { ExternalLink } from 'theme'
+import { calculateGasMargin } from 'utils'
+import { formatDollarAmount } from 'utils/numbers'
+
+import { FeeTag } from './ElasticFarmGroup/styleds'
+
+const Wrapper = styled.div`
+  line-height: 1.5;
+  border: 1px solid ${({ theme }) => theme.border};
+  background: ${({ theme }) => theme.background};
+  padding: 24px;
+  text-align: center;
+  font-size: 14px;
+  color: ${({ theme }) => theme.subText};
+  border-radius: 20px;
+
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    padding: 1rem 0.75rem;
+  `}
+`
+const TableHeader = styled.div`
+  display: grid;
+  font-size: 12px;
+  border-top-left-radius: 8px;
+  border-top-right-radius: 8px;
+  grid-template-columns: 2fr 3fr 3fr;
+  font-weight: 500;
+  padding: 1rem;
+  background: ${({ theme }) => theme.buttonBlack};
+  color: ${({ theme }) => theme.subText};
+  align-items: center;
+`
+const TableRow = styled(TableHeader)`
+  background: ${({ theme }) => theme.background};
+  color: ${({ theme }) => theme.text};
+  border-bottom: 1px solid ${({ theme }) => theme.border};
+`
+
+export const ElasticCompensation = ({
+  onWithdraw,
+  data,
+  numberOfPosition,
+  tokenPrices,
+  claimInfo,
+}: {
+  onWithdraw: () => void
+  data: Array<Snapshot> | null
+  numberOfPosition: number
+  tokenPrices: { [address: string]: number }
+  claimInfo: { address: string; encodedData: string } | null
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const theme = useTheme()
+  const { chainId } = useActiveWeb3React()
+  const { library } = useWeb3React()
+
+  const { farms } = useElasticFarms()
+
+  const enrichData = data?.map(item => {
+    const pool = farms?.find(f => f.id === item.farm_address)?.pools.find(p => +p.pid === +item.pid)
+    return {
+      ...item,
+      pending_rewards: item.pending_rewards.map(rw => {
+        const rwToken =
+          rw.token_address === ZERO_ADDRESS
+            ? NativeCurrencies[chainId]
+            : pool?.rewardTokens.find(token => token.wrapped.address.toLowerCase() === rw.token_address)
+        return {
+          amount: rw.amount,
+          token: rwToken,
```

Comment:
rwToken can be undefined

Reply:
undefined is okay, I checked it when using